src/figuresClasses.ts

```diff
@@ -1,11 +1,89 @@
-export interface Figure {}
+type Shape = `triangle` | `circle` | `rectangle`;
+type Color = `red` | `green` | `blue`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
[CODE KNOWLEDGE] - You don't need to manually assign the constructor parameters to properties if you're using TypeScript's parameter properties feature. Simply declaring them as public in the constructor signature is enough. Remove the manual assignments to keep the code clean.